snuba/query/processors/uuid_column_processor.py

```diff
@@ -26,6 +27,141 @@
 metrics = MetricsWrapper(environment.metrics, "api.query.uuid_processor")
 
 
+class UUIDTranslationError(Exception):
+    pass
+
+
+class BetterUUIDColumnProcessor(QueryProcessor):
+    """
+    Processor that handles columns which are stored as UUID in ClickHouse but
+    are represented externally as a non hyphenated hex string.
+    """
+
+    def __init__(self, uuid_columns: Set[str]) -> None:
+        self.__uuid_columns = uuid_columns
+        self.__uuid_column_match = Or([String(col) for col in uuid_columns])
+
+        operator = Param(
+            "operator",
+            Or(
+                [
+                    String(op)
+                    for op in FUNCTION_TO_OPERATOR
+                    if op not in (ConditionFunctions.IN, ConditionFunctions.NOT_IN)
+                ]
+            ),
+        )
+
+        in_operators = Param(
+            "operator",
+            Or((String(ConditionFunctions.IN), String(ConditionFunctions.NOT_IN))),
+        )
+
+        col_func = FunctionCallMatch(
+            String("replaceAll"),
+            (
+                FunctionCallMatch(
+                    String("toString"),
+                    (Param("col", ColumnMatch(None, self.__uuid_column_match)),),
+                ),
+            ),
+            with_optionals=True,
+        )
+
+        literal = Param("literal", LiteralMatch(AnyMatch(str)))
+
+        self.__condition_matcher = Or(
+            [
+                FunctionCallMatch(operator, (literal, col_func)),
+                FunctionCallMatch(operator, (col_func, literal)),
+            ]
+        )
```

Comment:
You can build a commutative condition pattern this way 
https://github.com/getsentry/snuba/blob/master/snuba/query/conditions.py#L144-L170

Reply:
Good to know that's there. I don't think it makes much sense in this case since the operator needs a named param.

Reply:
mh, right. We should change that.